### ISSUE TITLE
Log metric row after validation passes

### ISSUE THREAD

**Issue by IvanLazarevsky** [williamFalcon/pytorch-lightning#185]
**Is your feature request related to a problem? Please describe.**
metrics.csv file, used with test-tube Experiment, is pretty nice way to report some simple results. Unfortunately, logging gets invoked only after training steps. As a result, I can't get the last validation pass results in this file. 

**Describe the solution you'd like**
Option to write metrics from validation_end in a file, similar to metrics.csv, as a Trainer parameter.

**Describe alternatives you've considered**
Logging directly in "validation_end" method, but I haven't found a way to get the context (Experiment, Trainer state) to enrich the information.
Perhaps some access to active Trainer from LightningModule functions, so we can control logging without global variables or duplicating this functionality.

**Comment by williamFalcon** [comment]
1. 
in any part of the LightningModule you can access self.experiment and self.trainer.  This includes validation_end. 

2. 
metrics.csv logs whatever you add to self.trainer.log(...). In addition, tensorboard metrics can be downloaded from any graph by clicking download as csv. 

is that what you mean?

**Comment by IvanLazarevsky** [comment]
1. Exactly. Feeling embarrassed not to check that.

**Comment by williamFalcon** [comment]
maybe the docs can be made more clear regarding this?

**Comment by expectopatronum** [comment]
> metrics.csv logs whatever you add to self.trainer.log(...). In addition, tensorboard metrics can be downloaded from any graph by clicking download as csv.
I'm confused - I can't find a log function in the trainer class - am I missing something?

**Comment by williamFalcon** [comment]
self.experiment.log
